### PR TITLE
Fix permission error for flutter_file_downloader on Android 13

### PR DESCRIPTION
**Pull request by Nhinlee**

### CODE REVIEW

**Review by abdallah-odeh — COMMENTED** [review]
Hello @Nhinlee,
Doing so might cause an error if the user did not ask for the permission before requesting to download a file, plus, I actually prefer to keep the plugin independent because some projects might not need the plugin and in your case they will have to add the plugin to their app to use this feature.
I will test the plugin on Android 13 to see if there is any errors or crashes
Thank you for helping to make this plugin better

**Comment by Nhinlee** [comment]
> Hello @Nhinlee, Doing so might cause an error if the user did not ask for the permission before requesting to download a file, plus, I actually prefer to keep the plugin independent because some projects might not need the plugin and in your case they will have to add the plugin to their app to use this feature. I will test the plugin on Android 13 to see if there is any errors or crashes Thank you for helping to make this plugin better

Yes thanks @abdallah-odeh, pls help me to update that logic when y found the issues 👏

File: android/src/main/java/com/example/flutter_file_downloader/MethodCallHandlerImpl.java [hunk 1]
```diff
@@ -157,18 +157,19 @@ public void onResult(StoragePermission permission) {
...
+        //                onRequestPermission(helper, false);
+        //                return;
+        //            }
+        //        } catch (PermissionUndefinedException e) {
+        //            helper.result.error(
+        //                    ErrorCodes.permissionDefinitionsNotFound.toString(),
+        //                    ErrorCodes.permissionDefinitionsNotFound.toDescription(),
+        //                    null);
+        //            return;
+        //        }
```
Comment (phamnhuvu-dev) [inline android/src/main/java/com/example/flutter_file_downloader/MethodCallHandlerImpl.java:172]: private void onStartDownloadingFile(StoreHelper helper, { boolean autoHandlePermission = true }) {
        if (autoHandlePermission) {
        try {
            if (!permissionManager.hasPermission(context)) {
                onRequestPermission(helper, false);
                return;
            }
        } catch (PermissionUndefinedException e) {
            helper.result.error(
                    ErrorCodes.permissionDefinitionsNotFound.toString(),
                    ErrorCodes.permissionDefinitionsNotFound.toDescription(),
                    null);
            return;
        }
        }
  }.

Comment (abdallah-odeh) [reply android/src/main/java/com/example/flutter_file_downloader/MethodCallHandlerImpl.java]: May I ask how exactly is this helpful?
What if the user set it to no?
The app might crash

Comment (abdallah-odeh) [reply android/src/main/java/com/example/flutter_file_downloader/MethodCallHandlerImpl.java]: May I ask how exactly is this helpful?
What if the user set it to no?
The app might crash

Comment (phamnhuvu-dev) [reply android/src/main/java/com/example/flutter_file_downloader/MethodCallHandlerImpl.java]: Why:
flutter_file_downloader use the logic permission handler & don’t work with Android 13 => he want to delegate for [permission_handler](https://pub.dev/packages/permission_handler) package do it instead.

Comment (Nhinlee) [reply android/src/main/java/com/example/flutter_file_downloader/MethodCallHandlerImpl.java]: I updated code @abdallah-odeh , This is helpful for someone need to handle permission by themself.
If user set autoHandlePermission = false & don't handle permission by themself, will have the logs & not crash app 🙏

Comment (Nhinlee) [reply android/src/main/java/com/example/flutter_file_downloader/MethodCallHandlerImpl.java]: Can y help me to reviews to @abdallah-odeh 👍

**Comment by abdallah-odeh** [comment]
Can you please tell me the issue details?
did you face an issue with android 13? 
I tested the plugin on android 13 and everything works perfectly!
Please add any error logs or issues you faced

**Comment by Nhinlee** [comment]
> Can you please tell me the issue details? did you face an issue with android 13? I tested the plugin on android 13 and everything works perfectly! Please add any error logs or issues you faced

Here is the bug I met @abdallah-odeh:

### Context: **Adjust compileSdkVersion to 33**
```
android {
    compileSdkVersion 33

    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }

    kotlinOptions {
        jvmTarget = '1.8'
    }

    sourceSets {
        main.java.srcDirs += 'src/main/kotlin'
    }

    defaultConfig {
        // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
        applicationId "com.manabie.example.test_new_feature"
        minSdkVersion 21
        targetSdkVersion 33
        versionCode flutterVersionCode.toInteger()
        versionName flutterVersionName
    }

    buildTypes {
        release {
            // TODO: Add your own signing config for the release build.
            // Signing with the debug keys for now, so `flutter run --release` works.
            signingConfig signingConfigs.debug
        }
    }
}

```

### Record video:
https://user-images.githubusercontent.com/48386426/209256891-12132d3d-a377-4125-ba90-689a508b9bb8.mov


### Logs:
`I/flutter ( 5650): downloadFile error: PlatformException(PERMISSION_DEFINITIONS_NOT_FOUND, No storage permission is defined in the manifest. Make sure that WRITE_EXTERNAL_STORAGE is defined in the manifest., null, null)`

**Comment by abdallah-odeh** [comment]
Hello @Nhinlee & @phamnhuvu-dev,
I hope you both having a great day, in the newer versions of android there is no need to request storage permission to download a file, can you please try out the new prerelease of the plugin `flutter_file_downloader: ^1.1.1-dev.2` and tell me if you faced any issue.
Thank you for your time & patience.

**Comment by Nhinlee** [comment]
> Hello @Nhinlee & @phamnhuvu-dev, I hope you both having a great day, in the newer versions of android there is no need to request storage permission to download a file, can you please try out the new prerelease of the plugin `flutter_file_downloader: ^1.1.1-dev.2` and tell me if you faced any issue. Thank you for your time & patience.

Very thanks @abdallah-odeh for your quick reply!
I tested and it's work well from my side.

**Comment by abdallah-odeh** [comment]
Great, thank you for your time @Nhinlee

**Comment by abdallah-odeh** [comment]
Closed this PR due to inactivity, please feel free to reopen it if you have anything further.

**Pull request closed without merging**